rasa/core/actions/forms.py

```diff
@@ -308,14 +312,22 @@ def _user_rejected_manually(self, validation_events: List[Event]) -> bool:
     def _get_events_since_last_user_uttered(
         tracker: "DialogueStateTracker",
     ) -> List[SlotSet]:
-        if tracker.latest_message in tracker.events:
-            index = tracker.events.index(tracker.latest_message)
-        else:
+        # TODO: Better way to get this latest_message index is through an instance
+        # variable, eg. tracker.latest_message_index
```

Comment:
I think I was wrong when I said we couldn't do this because of the tracker store. We could probably use a similar method to `latest_message`, i.e. setting it in `UserUttered.apply_to`. When we deserialize a tracker from a store we re-apply each event in order, so it should work

Reply:
@joejuzl Yes, this is something I discussed with @ancalita as well. The naming would be a little odd, we not only want latest message index but also `restart` event. I can create another PR to check this alternative if that makes sense?

Reply:
Yeah - fine to look at it in a separate issue/PR I think.